tasks/Module 5/task_2.py

```diff
@@ -0,0 +1,231 @@
+"""Hero's Adventure Game."""
+from random import randint, choice
+from dataclasses import dataclass
+from task_1 import Item, Inventory, LIGHT_WEIGHT_THRESHOLD
+
+
+FOOD = ["apple", "wine", "chicken", "bread"]
+RUBBISH = ["chewed gum", "used tissue"]
+ITEMS = ["gold coin", "ruby", "dagger", "rope", "sword", "shield"] + FOOD + RUBBISH
+MOVES = ["north", "south", "west", "east", "l", "r", "u", "d"]
+
+
+@dataclass
+class GameItem(Item):
+    """GameItem class."""
+
+    heal: int = 0
+
+    def is_rubbish(self):
+        """Check if Item is rubbish."""
+        return self.name in RUBBISH
+
+    def is_food(self):
+        """Check if Item is food."""
+        return self.name in FOOD
+
+
+class Hero:
+    """Class Hero."""
+
+    def __init__(self, row, columm):
+        """Init Hero."""
+        super().__init__()
+        self.inv = Inventory()
+        self.stamina = 50
+        self.position = (row - 1, columm - 1)
+        self.can_move = True
+
+    def set_position(self, row, colum):
+        """Set initial posision of hero on a field."""
+        self.position = (row, colum)
+
+    def heal(self, item):
+        """Heal the hero with food."""
+        self.stamina += item.heal
+        print(
+            f"I got the {item.name}. Increased stamina with {item.heal}."
+            " Current stamia is {self.stamina}"
+        )
+
+    def get_stats(self):
+        """Print hero statistics."""
+        print(f"Stamina {self.stamina}/50\n" f"{self.inv}")
+
+    def drop_item(self, item):
+        """Remove item vrom inventory."""
+        self.inv.remove_item(item)
+
+    def check_heal(self):
+        """Check current stamina value."""
+        if self.stamina <= 0:
+            print(f"You fainted and die, your stamina is {self.stamina}")
+            return -1
+        if self.stamina <= 10:
+            print(f"Your healt is too low - {self.stamina}, you need more food")
+            return 0
+        return 1
+
+    def check_weight(self):
+        """Check weight of backpack and ask for drop."""
+        if not self.can_move:
+            print(
+                f"You are overweighed. Current weight of your backpack is {self.inv.get_weight()}\n"
+                f"Please drop any item from Inventory"
+            )
+            self.get_stats()
+            responce = input("Write an item name to drop: ").rstrip()
+            self.drop_item(responce)
+        if self.inv.check_weight() < 3:
+            self.can_move = True
+        return self.can_move
+
+
+class Field:
```

Comment:
Name of this class is misleading IMO. It represents the whole board, not a single field, so maybe` class Board: `would be more appropriate.